docs/reference/esql/processing-commands/keep.asciidoc

```diff
@@ -40,10 +50,84 @@ include::{esql-specs}/docs.csv-spec[tag=keepWildcard]
 ----
 
 The asterisk wildcard (`*`) by itself translates to all columns that do not
-match the other arguments. This query will first return all columns with a name
+match the other arguments.
+
+This query will first return all columns with a name
 that starts with `h`, followed by all other columns:
 
 [source,esql]
 ----
 include::{esql-specs}/docs.csv-spec[tag=keepDoubleWildcard]
 ----
+
+The following examples show how precedence rules work when a field name matches multiple expressions.
+
+Full name duplicates, last one wins:
+
+[source,esql]
+----
+include::{esql-specs}/docs.csv-spec[tag=keepMultipleMatch]
+----
+[%header.monospaced.styled,format=dsv,separator=|]
+|===
+include::{esql-specs}/docs.csv-spec[tag=keep-multiple-match-result]
+|===
+
+Complete field name has precedence over wildcard expressions:
+
+[source,esql]
+----
+include::{esql-specs}/docs.csv-spec[tag=keepCompleteName]
+----
+[%header.monospaced.styled,format=dsv,separator=|]
+|===
+include::{esql-specs}/docs.csv-spec[tag=keep-complete-name-result]
+|===
+
+Wildcard expressions have the same priority, but last one wins (despite being less specific):
+
+[source,esql]
+----
+include::{esql-specs}/docs.csv-spec[tag=keepWildcardPrecedence]
+----
+[%header.monospaced.styled,format=dsv,separator=|]
+|===
+include::{esql-specs}/docs.csv-spec[tag=keep-wildcard-precedence-result]
+|===
+
+// row foo = 1, bar = 2 | keep *, foo   ->  bar, foo
+
+A simple wildcard expression `*` has the lowest priority.
+Output order is determined by the other arguments:
+
+[source,esql]
+----
+include::{esql-specs}/docs.csv-spec[tag=keepWildcardLowest]
+----
+[%header.monospaced.styled,format=dsv,separator=|]
+|===
+include::{esql-specs}/docs.csv-spec[tag=keep-wildcard-lowest-result]
+|===
+
+[source,esql]
+----
+include::{esql-specs}/docs.csv-spec[tag=keepWildcardLowest2]
+----
+[%header.monospaced.styled,format=dsv,separator=|]
+|===
+include::{esql-specs}/docs.csv-spec[tag=keep-wildcard-lowest-result2]
+|===
+
+[source,esql]
+----
+include::{esql-specs}/docs.csv-spec[tag=keepWildcardLowest3]
+----
+[%header.monospaced.styled,format=dsv,separator=|]
+|===
+include::{esql-specs}/docs.csv-spec[tag=keep-wildcard-lowest-result3]
+|===
```

Comment:
I feel like these two examples are already covered by `KEEP h*, *`; maybe we shouldn't add them, to keep the examples focused?